mpmp/src/model/Player.java

```diff
@@ -6,366 +6,368 @@
-		pos = 0;
-		cash = StartCash;
-		hyp = 0;
-	    }
-	}
+	// XXX move to a new file of constants?
+	private static final int StartCash = 30000; // XXX value
+	private static final int Wage = 4000;  // XXX value
+	private static final int IncomeTax = 4000;  // XXX value
+	private static final int ExtraTax = 2000;  // XXX value
+	public static final int UnjailFee = 1000;
+	private static Player currentPlayer;
 
-	if (mode == Mode.Player) {
-	    plots = new HashSet<>();
-	    hypothecs = new HashSet<>();
-	    pos = 0;
-	    cash = StartCash;
-	    hyp = 0;
-	}
-    }
+	public enum Mode {
 
-    public String getName() {
-	return name;
-    }
+		Spectator, Player
+	}
 
-    public Color getColor() {
-	return color;
-    }
+	/* common to spectators and players */
+	private String name;
+	private Color color;
+	private Mode mode;
+	private static ArrayList<Player> players;
+
+	/* only active players */
+	private HashSet<Plot> plots;
+	private HashSet<Plot> hypothecs;
+	private int cash;
+	/* actual liquid money */
+	private int hyp;
+	/* hypothec money */
+	private int pos;
+	/* 0 is start; counted clockwise */
+	private boolean inPrison;
+	private int unjails;
+
+	/* number of unjail cards */
+	public Player(Color color, Mode mode, String name) {
+		this.name = name;
+		this.color = color;
+		this.mode = mode;
+
+		synchronized (players) {
+			players.add(this);
+
+			if (mode == Mode.Player) {
+				plots = new HashSet<>();
+				hypothecs = new HashSet<>();
+				pos = 0;
+				cash = StartCash;
+				hyp = 0;
+			}
+		}
 
-    public int getPos() {
-	return pos;
-    }
+		if (mode == Mode.Player) {
+			plots = new HashSet<>();
+			hypothecs = new HashSet<>();
+			pos = 0;
+			cash = StartCash;
+			hyp = 0;
+		}
+	}
 
-    public int getMoney() {
-	return cash;
-    }
+	public String getName() {
+		return name;
+	}
 
-    public String getPlots() {
-	return plots.toString();
-    }
+	public Color getColor() {
+		return color;
+	}
 
-    public boolean isInJail() {
-	return inPrison;
-    }
+	public int getPos() {
+		return pos;
+	}
 
-    public String toString() {
-	// Convert to hex triplet without alpha value and in uppercase.
-	String col = "#" + Integer.toHexString(color.getRGB()).substring(2).toUpperCase();
-	return col + ": " + mode + ": " + name;
-    }
+	public int getMoney() {
+		return cash;
+	}
 
-    public void remove() {
-	synchronized (players) {
-	    players.remove(this);
+	public String getPlots() {
+		return plots.toString();
 	}
-    }
-
-    /* GAME LOGIC */
-    /**
-     * Adds or removes money from a player. If the player can't pay, false is returned.
-     */
-    public boolean addMoney(int sum) {
-	// XXX simply allow negative money instead of returning right here might be better. -oki
-
-	/* Sum is often negative. Have we positive money if we add sum? */
-	if (cash + hyp + sum < 0) {
-	    return false;
+
+	public boolean isInJail() {
+		return inPrison;
 	}
 
-	cash += sum;
+	public String toString() {
```

Comment:
@leletec A missing `Override` annotation is really unimportant. The compiler doesn't need it.
@dieechtenilente Please check that you're not accidentally changing `Player.java` to an earlier state.